src/test/clojure/xtdb/trie_test.clj

```diff
@@ -138,6 +138,22 @@
                                             temporal-bounds)
                     (merge-plan-nodes->path+pages))))))))
 
+(defn ->test-merge-plan-node [page metadata-matches?]
+  (reify MergePlanPage
+    (load-page [_ _ _] page)
```

Comment:
might be worth throwing UOE here, make it clear that this isn't supported for this implementation and confirm that we're not accidentally using it in tests that try to load the page

Reply:
But I am using it in the test just below. Essentially identifying the page (page + segment in the tests to come).